src/path.ts

```diff
@@ -0,0 +1,12 @@
+import * as path from 'path';
+import * as URI from 'urijs';
+import { URI as VSURI } from 'vscode-uri';
+
+export const isAbsolute = (filepath: string) => path.isAbsolute(filepath) || new URI(filepath).is('absolute');
+
+export const join = (...parts: string[]) => parts.join('/');
+export const normalize = (filepath: string) => path.normalize(filepath).replace(/\\/g, '/');
```

Comment:
What happens when non file e.g. `https://localhost:3000/bar.json?page=1#hello` is pass into all of these helpers? Can you add a few test cases for this?